src/emulator/storage/server.ts

```diff
@@ -119,3 +162,11 @@ export function createApp(
 
   return Promise.resolve(app);
 }
+
+type RulesFile = { name: string; content: string; resource?: string };
+
+function isRulesFile(file: unknown): file is RulesFile {
```

Comment:
why can't you replace this entire function with `obj instanceof RulesFile`?

Reply:
Because `RulesFile` is a type alias, not a class. Btw, I'm not too familiar with type predicates either, just following https://google.github.io/styleguide/tsguide.html#any-unknown.